diff --git a/networkx/algorithms/tree/mst.py b/networkx/algorithms/tree/mst.py
index 984a38b74ab..b06412598f2 100644
--- a/networkx/algorithms/tree/mst.py
+++ b/networkx/algorithms/tree/mst.py
@@ -1074,24 +1074,9 @@ def __init__(self, G, weight="weight", minimum=True, ignore_nan=False):
         self.partition_key = (
             "SpanningTreeIterators super secret partition attribute name"
         )
+        self.partition_queue = None
 
     def __iter__(self):
-        """
-        Returns
-        -------
-        SpanningTreeIterator
-            The iterator object for this graph
-        """
-        self.partition_queue = PriorityQueue()
-        self._clear_partition(self.G)
-        mst_weight = partition_spanning_tree(
-            self.G, self.minimum, self.weight, self.partition_key, self.ignore_nan
-        ).size(weight=self.weight)
-
-        self.partition_queue.put(
-            self.Partition(mst_weight if self.minimum else -mst_weight, {})
-        )
-
         return self
 
     def __next__(self):
@@ -1102,6 +1087,15 @@ def __next__(self):
             The spanning tree of next greatest weight, which ties broken
             arbitrarily.
         """
+        if self.partition_queue is None:
+            self.partition_queue = PriorityQueue()
+            self._clear_partition(self.G)
+            mst_weight = partition_spanning_tree(
+                self.G, self.minimum, self.weight, self.partition_key, self.ignore_nan
+            ).size(weight=self.weight)
+            self.partition_queue.put(
+                self.Partition(mst_weight if self.minimum else -mst_weight, {})
+            )
         if self.partition_queue.empty():
             del self.G, self.partition_queue
             raise StopIteration
