diff --git a/faker/proxy.py b/faker/proxy.py
index 758e9bf83b..b9506c83aa 100644
--- a/faker/proxy.py
+++ b/faker/proxy.py
@@ -11,7 +11,7 @@
 from .config import DEFAULT_LOCALE
 from .exceptions import UniquenessException
 from .factory import Factory
-from .generator import Generator, random
+from .generator import Generator
 from .typing import SeedType
 from .utils.distribution import choices_distribution
 
@@ -183,10 +183,10 @@ def _select_factory(self, method_name: str) -> Factory:
         return factory
 
     def _select_factory_distribution(self, factories, weights):
-        return choices_distribution(factories, weights, random, length=1)[0]
+        return choices_distribution(factories, weights, self.factories[0].random, length=1)[0]
 
     def _select_factory_choice(self, factories):
-        return random.choice(factories)
+        return self._factories[0].random.choice(factories)
 
     def _map_provider_method(self, method_name: str) -> tuple[list[Factory], list[float] | None]:
         """
